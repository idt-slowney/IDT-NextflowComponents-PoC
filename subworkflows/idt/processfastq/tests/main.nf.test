// TODO nf-core: Once you have added the required tests, please run the following command to build this file:
// nf-core subworkflows test processfastq
nextflow_workflow {

    name "Test Subworkflow PROCESSFASTQ"
    script "../main.nf"
    workflow "PROCESSFASTQ"

    tag "subworkflows"
    tag "subworkflows_nfcore"
    tag "subworkflows/processfastq"
    // TODO nf-core: Add tags for all modules used within this subworkflow. Example:
    tag "fastqc"
    tag "seqtk/sample"
    tag "cutadapt/extractumi"
    tag "trimgalore"


    // TODO nf-core: Change the test name preferably indicating the test-data and file-format used
    test("homo_sapiens - fastq - paired_end") {

        when {
            workflow {
                """
                // TODO nf-core: define inputs of the workflow here. Example:
                input[0] = [ [ id:'test', single_end:false, sample_size:5000 ], // meta map
                        [file(params.test_data['homo_sapiens']['illumina']['test_1_fastq_gz'], checkIfExists: true),
                         file(params.test_data['homo_sapiens']['illumina']['test_2_fastq_gz'], checkIfExists: true)
                        ]
                input[1] = [ false ]
                input[2] = [ false ]
                input[3] = [ false ]
                input[4] = [ true ]
                input[5] = [ test ]
                input[6] = [ false ]
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(workflow.out).match()}
                //TODO nf-core: Add all required assertions to verify the test output.
            )
        }
    }
}
