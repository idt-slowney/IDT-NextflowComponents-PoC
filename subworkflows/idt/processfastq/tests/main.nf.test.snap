{
    "homo_sapiens - fastq - paired_end": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false,
                        "sample_size": 5000
                    },
                    [
                        "/nf-core/test-datasets/modules/data/genomics/homo_sapiens/illumina/fastq/test_1.fastq.gz",
                        "/nf-core/test-datasets/modules/data/genomics/homo_sapiens/illumina/fastq/test_2.fastq.gz"
                    ]
                ]
            ],
            [
                "versions.yml:md5,7b9f999b76de4a7d95267dfd2d9ddec0"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-08-06T10:28:36.852616387"
    }
}